lib/internal/fs/promises.js

```diff
@@ -346,7 +346,16 @@ async function open(path, flags, mode) {
                                  flagsNumber, mode, kUsePromises));
 }
 
-async function read(handle, buffer, offset, length, position) {
+async function read(handle, bufferOrOptions, offset, length, position) {
+  let buffer = bufferOrOptions;
+
+  if (!isArrayBufferView(buffer)) {
+    buffer = bufferOrOptions.buffer || Buffer.alloc(16384);
+    offset = bufferOrOptions.offset || 0;
+    length = buffer.length;
+    position = bufferOrOptions.position || null;
+  }
+
   validateBuffer(buffer);
```

Comment:
```suggestion
  let buffer = bufferOrOptions;

  if (!isArrayBufferView(buffer)) {
    if(bufferOrOptions.buffer) {
      buffer = bufferOrOptions.buffer
      validateBuffer(buffer);
    } else {
      buffer = Buffer.alloc(16384);
    }
    offset = bufferOrOptions.offset || 0;
    length = buffer.length;
    position = bufferOrOptions.position || null;
  }
```

Reply:
Thank you @aduh95. This way, double validation is done only when buffer is object, and contains buffer property.